File: plugins/kotlin/j2k/new/tests/testData/newJ2k/typeCastExpression/doubleCasts.java
class A {
    void doubleToChar() {
        char a = (char) 1.0;
        double b = 2;
        char c = (char) b;
        char d = (char) (b + c);
    }

    void doubleToByte() {
        byte a = (byte) 1.0;
        double b = 2;
        byte c = (byte) b;
        byte d = (byte) (b + c);
    }

    void doubleToShort() {
        short a = (short) 1.0;
        double b = 2;
        short c = (short) b;
        short d = (short) (b + c);
    }

    void doubleToInt() {
        int a = (int) 1.0;
        double b = 2;
        int c = (int) b;
        int d = (int) (b + c);
    }

    void doubleToLong() {
        long a = (long) 1.0;
        double b = 2;
        long c = (long) b;
        long d = (long) (b + c);
    }

    void doubleToFloat() {
        float a = (float) 1.0;
        double b = 2;
        float c = (float) b;
        float d = (float) (b + c);
    }

    void charToDouble() {
        double a = 'a';
        char b = 'b';
        double c = b;
        double d = b + c;
    }

    void byteToDouble() {
        double a = (byte) 1;
        byte b = 2;
        double c = b;
        double d = b + c;
    }

    void shortToDouble() {
        double a = (short) 1;
        short b = 2;
        double c = b;
        double d = b + c;
    }

    void intToDouble() {
        double a = 1;
        int b = 2;
        double c = b;
        double d = b + c;
    }

    void longToDouble() {
        double a = (double) 1L;
        long b = 2;
        double c = (double) b;
        double d = (double) (b + c);
    }

    void floatToDouble() {
        double a = 1.0f;
        float b = 2;
        double c = b;
        double d = b + c;
    }
}

File: plugins/kotlin/j2k/shared/tests/testData/newJ2k/typeCastExpression/doubleCasts.kt
internal class A {
    fun doubleToChar() {
        val a = 1.0.toInt().toChar()
        val b = 2.0
        val c = b.toInt().toChar()
        val d = (b + c.code.toDouble()).toInt().toChar()
    }

    fun doubleToByte() {
        val a = 1.0.toInt().toByte()
        val b = 2.0
        val c = b.toInt().toByte()
        val d = (b + c).toInt().toByte()
    }

    fun doubleToShort() {
        val a = 1.0.toInt().toShort()
        val b = 2.0
        val c = b.toInt().toShort()
        val d = (b + c).toInt().toShort()
    }

    fun doubleToInt() {
        val a = 1.0.toInt()
        val b = 2.0
        val c = b.toInt()
        val d = (b + c).toInt()
    }

    fun doubleToLong() {
        val a = 1.0.toLong()
        val b = 2.0
        val c = b.toLong()
        val d = (b + c).toLong()
    }

    fun doubleToFloat() {
        val a = 1.0.toFloat()
        val b = 2.0
        val c = b.toFloat()
        val d = (b + c).toFloat()
    }

    fun charToDouble() {
        val a = 'a'.code.toDouble()
        val b = 'b'
        val c = b.code.toDouble()
        val d = b.code.toDouble() + c
    }

    fun byteToDouble() {
        val a = 1.toByte().toDouble()
        val b: Byte = 2
        val c = b.toDouble()
        val d = b + c
    }

    fun shortToDouble() {
        val a = 1.toShort().toDouble()
        val b: Short = 2
        val c = b.toDouble()
        val d = b + c
    }

    fun intToDouble() {
        val a = 1.0
        val b = 2
        val c = b.toDouble()
        val d = b + c
    }

    fun longToDouble() {
        val a = 1.0
        val b: Long = 2
        val c = b.toDouble()
        val d = (b + c)
    }

    fun floatToDouble() {
        val a = 1.0
        val b = 2f
        val c = b.toDouble()
        val d = b + c
    }
}
